snap/implicit.go

```diff
@@ -62,6 +62,7 @@ var implicitClassicSlots = []string{
 	"browser-support",
 	"cups-control",
 	"gsettings",
+	"i2c",
```

Comment:
That doesn't make much sense I think. How can it be implicit if it requires a path attribute, and how can it be implicit in the core if the interface forces it to be in the gadget?

The interface name is also wrong per the current code.